### PR TITLE
Feature/integration specs

### PR DESCRIPTION
**Pull request by blessedjasonmwanza**
##Feature/integration specs
In this UX simulation integrations tests update;

- Follo test cases implementation [guidelines](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Integration_specs_project.md)
- Collaborate with @Saadat123456 to implement test cases
- Use Capybara to write integration tests for each view in 
  - [x] Login page:
  - [x] User index page
  - [x] user show page:
  - [x] User post index page
  - [x] Post show page

### CODE REVIEW

File: test/integration/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    end
+    it 'test the number of comments' do
+      expect(page).to have_content 'Comments 3'
+    end
+    it 'how many likes a post has.' do
+      expect(page).to have_content 'Likes 0'
+    end
+    it 'when i click post it redirects to post page' do
+      click_link 'post 1'
+      expect(page).to have_current_path(user_post_path(@user, @post1))
```
Comment (ahmetbozaci) [inline test/integration/post_index_spec.rb:69]: - [x] Kindly add a test for the user's profile picture. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Integration_specs_project.md#:~:text=post%20index%20page%3A-,I%20can%20see%20the%20user%27s%20profile%20picture.,-I%20can%20see).

File: test/integration/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      end
+      click_button 'Log in'
+      name = find_all('h2 a')
+      name[0].click
+      expect(page).to have_content('post 1')
+      expect(page).to have_content('post 2')
+      expect(page).to have_content('post 3')
+      expect(page).to have_link('See all posts')
+    end
+  end
```
Comment (ahmetbozaci) [inline test/integration/user_show_spec.rb]: - [x] Kindly add a test for the user's profile picture. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Integration_specs_project.md#:~:text=user%20show%20page%3A-,I%20can%20see%20the%20user%27s%20profile%20picture.,-I%20can%20see).

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @blessedjasonmwanza  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- All tests that you wrote passed well done. ✔️
image

___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] Kindly fix the error.


image

___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: test/integration/post_index_spec.rb (hunk 1 above)
Comment (blessedjasonmwanza) [reply test/integration/post_index_spec.rb:69]: Actioned.

File: test/integration/user_show_spec.rb (hunk 2 above)
Comment (blessedjasonmwanza) [reply test/integration/user_show_spec.rb]: Actioned.

**Review by DammyShittu — APPROVED** [review]
Hi @blessedjasonmwanza and @Saadat123456,

## APPROVED 🟢

Great implementation of all the requirements in this milestone. 👏

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**